File: durak-client-angular/src/app/models/events/AddRoomEvent.ts
export class AddRoomEvent {
    connectionId: string;
    roomId: string;
    bet: number;
    playerCount: number;

    constructor(connectionId: string, roomId: string, bet: number, playerCount: number) {
        this.connectionId = connectionId;
        this.roomId = roomId;
        this.bet = bet;
        this.playerCount = playerCount;
    }
}

File: durak-server-orleans/Durak.Server.API/Hubs/RoomListEvents/AddRoomEvent.cs
using Durak.Server.API.Enums;
using Durak.Server.API.Models;

namespace Durak.Server.API.Hubs.RoomListEvents;

public class AddRoomEvent(string connectionId, RoomSettings roomSettings, byte playerCount) : IEvent
{
    public string MethodName { get; } = "AddRoom";

    public string ConnectionId { get; set; } = connectionId;
    public Guid RoomId { get; set; } = roomSettings.Id;
    public Bet Bet { get; set; } = roomSettings.Bet;
    public byte PlayerCount { get; set; } = playerCount;
}
